### ISSUE TITLE
Use only validation dataset with Detection Validator

### ISSUE THREAD

**Issue by maianumerosky** [ultralytics/ultralytics#4313]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I use a DetectionValidator class in order to retrieve metrics about my model, and I set the dataloader like so:
`dataloader = validator.get_dataloader(<PATH_TO_DATASET>, BATCH_SIZE)`; is there a way to get the validator to use only the validation dataset?

My folder structure is:
```
<PATH_TO_DATASET>
|_ train
    |_ images
    |_ labels
|_ val
    |_ images
    |_ labels
|_ data.yaml
```

and data.yaml contains the paths to the train and val dataset.
But I don't want to delete the train folder in order to run the validator metrics, I just want to "skip" it.
Thanks in advance!

**Comment by glenn-jocher** [comment]
@maianumerosky validation is not performed on training data, thus there is nothing to "skip", just validate normally, i.e. `yolo val data=coco8.yaml`

**Comment by maianumerosky** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/17316848/b760b9a5-8c36-442c-bfd3-11a49587919c)
Hi @glenn-jocher, thanks for your answer. If I run the code in the image without deleting my `train` folder, the dataloader ends up being populated with all of the images (not only validation but training as well) . As the dataloader does not store info on whether it's used for training or for validation, when performing benchmarking and one iterates along batches of the dataloader, one ends up performing validation on all of the data (the training set as well).
I believe the only way to avoid this is to skip the training data when calling `validator.data = check_det_dataset(args.data)
`, since this looks for a YAML file with dirs for both the train and val datasets.

**Comment by glenn-jocher** [comment]
@maianumerosky hi, thanks for your question.

Your understanding is partially correct. The `data` attribute of the `DetectionValidator` class does indeed look at the YAML file and compiles a list of all image and label files in both your training and validation directories. However, during the validation phase, only the validation dataset is actually used.

The purpose of including both training and validation data is to ensure data uniformity in terms of classes and format. This does not mean that the model validates on the training data.

Even though the dataloader may seem to be populated with all images, metrics computation in the validation phase only considers images from the validation set, not the training set. Therefore, you don't need to worry about skipping the training dataset; it is not evaluated during the validation phase, despite it appearing in the dataloader.
